src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -592,7 +592,8 @@ def parse_error_handler(parse_error):
         raise TaskError('Error parsing xml file at {}: {}'
           .format(parse_error.xml_path, parse_error.cause))
 
-      all_tests_info = self.parse_test_info(junitxml_path, parse_error_handler, ['file', 'name'])
+      all_tests_info = self.parse_test_info(junitxml_path, parse_error_handler,
+                                            ['file', 'name', 'classname'])
```

Comment:
Will this do the right thing if no classname is actually reported in the XML? Namely: succeed, but skip recording that field? Can you add a test to confirm?